bundles/org.eclipse.packagedrone.utils.rpm/src/org/eclipse/packagedrone/utils/rpm/parse/RpmInputStream.java

```diff
@@ -132,16 +134,21 @@ private InputStream setupPayloadStream () throws IOException
 
         switch ( payloadCoding )
         {
-            case "none":
```

Comment:
I think I saw `none` at some point. What is the reason to remove this?

Reply:
The first commit I made to this pull request handled the case of "none" meaning no compression, following what you originally had in the code.

However, when I looked closer, it seems that RPM handles a missing compression flag in the header as defaulting to gzip. Therefore, it seems that if the payload coding is missing or empty, that is is supposed to default to gzip and not no compression. Also, it seems that it really shouldn't be possible to create RPMs without some compression, or it breaks spec.

Can you verify this? If it turns out not to be the case, then we can see about adding this back.

Reply:
I think both variants are wrong :grin:  -> https://github.com/rpm-software-management/rpm/blob/ff4b9111aeba01dd025dd133ce617fb80f7398a0/build/pack.c#L281

To me it looks like the correct value would be `null`.

Reply:
IIRC this was my source: http://rpm.org/devel_doc/file_format.html#24x-payload-codings

Reply:
Actually, the original code likely set the value to `"none"` on a `null` simply because `switch` doesn't support a `null` case. It's a bit confusing since `"none"` (if it were actually set in the header) is not a valid compressor. I will put the code back to allow creating uncompressed RPMs.

However, I am trying to see if the reading is consistent with this. For example, 
https://github.com/rpm-software-management/rpm/blob/c7e68754584c49764c397cbccc23aebc491b1a3a/lib/rpmte.c#L632

Is it true that a `null` value defaults to gzip? I think this is where I got that idea that it could not read uncompressed RPMs, but I should have just tried it.

Reply:
In the document you referenced, it states:

> If the payload coding is not specified via the 1125 (Payload coding) header entry, then it is assumed to be “gzip”. The “passthrough” mode feature of zlib is used to allow the lack of the 1125 (Payload coding) header entry to also represent “none”.

So, maybe that is how it get away with using gzip? In any case, this string "none" does not seem to appear anywhere in the source code.

Additionally, according to the documentation, we apparently need to distinguish between lack of a header and a header value of `null`. However, the C code does not seem to distinguish between these as a single statement seems to check for both the header existence and the value at once.

Reply:
> So, maybe that is how it get away with using gzip? In any case, this string "none" does not seem to appear anywhere in the source code.

It looks as if when rpm writes a file a missing header always indicates no compression. But, the way the reading is implemented, it may be that a missing header can contain either no compression or gzip compression. If this is the case, it should try gzip first and then fallback to no compression if not in gzip format. But, I'd really like to see this behavior verified against the official rpm first.